src/huggingface_hub/serialization/_dduf.py

```diff
@@ -0,0 +1,289 @@
+import logging
+import mmap
+import shutil
+import zipfile
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, Generator, Iterable, Tuple, Union
+
+from ..errors import DDUFCorruptedFileError, DDUFExportError
+
+
+logger = logging.getLogger(__name__)
+
+DDUF_ALLOWED_ENTRIES = {
+    ".gguf",
+    ".json",
+    ".model",
+    ".safetensors",
+    ".txt",
+}
```

Comment:
I am okay with this setup but just also saying as an FYI that `torchao` currently doesn't support `safetensors` serialization. So, any `diffusers` repo serialized with `torchao` won't make the cut here. But we can probably worry about that later.

Reply:
Might be a good incentive for them to support safetensors :)